src/Ens.ts

```diff
@@ -97,14 +139,71 @@ export default class Ens extends EthereumNamingService {
       return null;
     }
 
-    const resolverContract = this.buildContract(
+    const resolverContract = new Contract(
       resolverInterface(resolverAddress, EthCoinIndex),
       resolverAddress,
+      this.provider
     );
 
     return await this.resolverCallToName(resolverContract, nodeHash);
   }
 
+  /**
+   * This was done to make automated tests more configurable
+   */
+  private resolverCallToName(resolverContract: Contract, nodeHash) {
+    return this.callMethod(resolverContract, 'name', [nodeHash]);
+  }
```

Comment:
I'm not sure that mocking this method actually helps us to test anything. It's practically the same as we'd mocked the `reverse` method and tested that the mock actually returned the result we expected.